### PR TITLE
Experiment status is now updated after HF finishes.

### LINKED ISSUE #4: Changing experiment state to finished in ExperimentRun
Right now we have `ExperimentRun` responsible for starting all hypgen processes. It also responsible for changing DAP `Experiment` status into error when any exception occurs. Unfortunately logic responsible for setting `Experiment` status into finished (when everything went fine and results are generated) is located somewhere else (inside `hyperflow` workflow?). 

My question is as follow: Is it possible to move logic responsible for setting `Experiment` status into finished into `ExperimentRun`?

/cc @ismop/hypgen-dev-team

### PR DESCRIPTION
**Pull request by mpawlik**
Marek, could you confirm that audit returns audited code result?:)

Fixes #4

### CODE REVIEW

**Comment by mkasztelnik** [comment]
@mpawlik yup it returns block value. I corrected specs and do some alignment for workflow execution failure - please take a look.

File: lib/hypgen/workflow.rb [hunk 1]
```diff
@@ -53,6 +53,8 @@ def run!
         output = pipe.read
 
         puts "Workflow execution output: #{output}"
+        pipe.close
+        $?.to_i
```
Comment (mpawlik) [inline lib/hypgen/workflow.rb]: why .to_i is here?

Comment (mkasztelnik) [inline lib/hypgen/workflow.rb]: @mpawlik why do you ask me? This is [your](https://github.com/ismop/hypgen/blame/experiment_completion/lib/hypgen/workflow.rb) change :smile_cat:

Comment (mpawlik) [inline lib/hypgen/workflow.rb]: @mkasztelnik yeah, my bad:( too much ruby for one day. This .to_is is converting a 'result object' to simple return code, so it's necessary there.

**Pull request merged**